src/Clapi/Types/Path.hs

```diff
@@ -114,3 +114,8 @@ typeNameP = TypeName <$> segP <*> (DAT.char qualSepChar >> segP)
 typeNameFromText :: MonadFail m => Text -> m TypeName
 typeNameFromText =
   either fail return . DAT.parseOnly (typeNameP <* DAT.endOfInput)
+
+parentPath :: Path -> Maybe Path
+parentPath p = case p of
```

Comment:
Not too important, but it might be nice to define this in terms of `Clapi.Types.Path.splitTail` to save converting back and forth to the `Maybe`:
```
parentPath = fmap fst . splitTail
```
Looking again at `splitTail`, I don't think it's particularly efficient either, as it does both `init` and `last`. If you can think of a nice way to do a `splitLast` on a list, that would be cool.

Reply:
I don't think I can beat `O(n)`, but you can't for a singly linked list, but I think not going over it twice will probably be a reasonable way to go.